dataplane/updater.h

```diff
@@ -0,0 +1,219 @@
+#pragma once
+
+#include <mutex>
+#include <nlohmann/json.hpp>
+#include <rte_malloc.h>
+
+#include "common.h"
+#include "common/limit.h"
+#include "hashtable.h"
+#include "lpm.h"
+#include "memory_manager.h"
+
+namespace dataplane
+{
+
+[[maybe_unused]] static uint32_t upper_power_of_two(const uint32_t value)
+{
+	/// @todo: use __builtin_clz
+
+	uint32_t result = 1;
+	while (result < value)
+	{
+		result <<= 1;
+		if (!result)
+		{
+			return 0;
+		}
+	}
+	return result;
+}
+
+[[maybe_unused]] static std::string to_hex(const void* pointer)
+{
+	char buffer[128];
+	snprintf(buffer, 128, "%p", pointer);
+	return buffer;
+}
+
+class updater_lpm4_24bit_8bit_id32
+{
+public:
+	using object_type = lpm4_24bit_8bit_id32_dynamic;
+
+	updater_lpm4_24bit_8bit_id32(const char* name,
+	                             dataplane::memory_manager* memory_manager,
+	                             const tSocketId socket_id) :
+	        name(name),
+	        memory_manager(memory_manager),
+	        socket_id(socket_id),
+	        pointer(nullptr)
+	{
+	}
+
+	eResult update(const std::vector<common::acl::tree_chunk_8bit_t>& values)
+	{
+		stats.extended_chunks_size = std::max((uint64_t)object_type::extended_chunks_size_min,
+		                                      values.size() / 16);
+
+		for (;;)
+		{
+			pointer = memory_manager->create<object_type>(name.data(),
```

Comment:
I think, extended_chunks_size should be encapsulated into the tree but not the updater

Reply:
The idea was that worker and lookup is not required extended_chunks_size